modules/users/manifests/roller/account.pp

```diff
@@ -0,0 +1,98 @@
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+class users::roller::account($username, $group, $grouplist, $home) {
+    include ::config
+    include needs_reboot
+
+    ##
+    # sanity checks
+
+    if ($username == '') {
+        fail('No roller username set')
+    }
+
+    ##
+    # create the user
+
+    case $::operatingsystem {
+        Darwin: {
+            # use our custom type and provider, based on http://projects.puppetlabs.com/issues/12833
+            # This should be replaced with 'user' once we are running a version of puppet containing the
+            # relevant fixes.
+            # NOTE: this user is *not* an Administrator.  All admin-level access is granted via sudoers.
+            case $::macosx_productversion_major {
+                '10.7': {
+                    if (secret('roller_pw_saltedsha512') == '') {
+                        fail('No roller password saltedsha512 set')
+                    }
+                    darwinuser {
+                        $username:
+                            gid      => $group,
+                            shell    => '/bin/bash',
+                            home     => $home,
+                            password => secret('roller_pw_saltedsha512'),
+                            comment  => 'roller',
+                            notify   => Exec['kill-roller-keychain'];
+                    }
+                    $user_req = Darwinuser[$username]
+                }
+                '10.8': {
+                    if (secret('roller_pw_pbkdf2') == '' or secret('roller_pw_pbkdf2_salt') == '') {
+                        fail('No roller password pbkdf2 set')
+                    }
+                    darwinuser {
+                        $username:
+                            shell      => '/bin/bash',
+                            home       => $home,
+                            password   => secret('roller_pw_pbkdf2'),
+                            salt       => secret('roller_pw_pbkdf2_salt'),
+                            iterations => secret('roller_pw_pbkdf2_iterations'),
+                            comment    => 'roller',
+                            notify     => Exec['kill-roller-keychain'];
+                    }
+                    $user_req = Darwinuser[$username]
+                }
+                '10.9','10.10': {
+                    if (secret('roller_pw_pbkdf2') == '' or secret('roller_pw_pbkdf2_salt') == '') {
+                        fail('No roller password pbkdf2 set')
+                    }
+                    user {
+                        $username:
+                            shell      => '/bin/bash',
+                            home       => $home,
+                            password   => secret('roller_pw_pbkdf2'),
```

Comment:
Copied from mozreview:
> The roller users should *NOT* have a password at all since roller should only be accessing it via ssh key. It will just become another password that will have to be rotated at some point.
5/2/2018 [:dividehex]

Reply:
Copied from mozreview:
> I'm researching how to set up users without passwords in osx. I'll update with a new patch once I have that tested and working.
[:dhouse]

Reply:
I've added a MatchUser entry to the sshd_config to prevent password login for the roller user. I did not find another good method to remove or avoid passwords on the roller user (removing it from the user appears to break things in osx).